core/api/src/tasks/profile/completeImport.ts

```diff
@@ -50,35 +51,46 @@ export class ProfileCompleteImport extends RetryableTask {
       }
     }
 
+    const profilePropertyRules = await ProfilePropertyRule.cached();
+    const mergedValues = {};
     const imports = await profile.$get("imports", {
       where: { profileUpdatedAt: null },
       order: [["createdAt", "asc"]],
     });
 
-    try {
-      for (const i in imports) {
-        const _import = imports[i];
-        await profile.addOrUpdateProperties(_import.data);
+    for (const i in imports) {
+      const data = imports[i].data;
+      for (const key in data) {
+        // only if we still have property
+        if (profilePropertyRules[key]) {
+          mergedValues[key] = data[key];
+        }
       }
+    }
 
+    try {
+      await profile.addOrUpdateProperties(mergedValues);
       await profile.updateGroupMembership();
 
-      const newProfileProperties = await profile.properties();
+      const newProfileProperties = this.simplifyProfileProperties(
+        await profile.properties()
+      );
       const newGroups = await profile.$get("groups");
+      const newGroupGuids = newGroups.map((g) => g.guid);
+      const now = new Date();
 
       for (const i in imports) {
         const _import = imports[i];
-        _import.newProfileProperties = this.simplifyProfileProperties(
-          newProfileProperties
-        );
-        _import.profileUpdatedAt = new Date();
-        _import.newGroupGuids = newGroups.map((g) => g.guid);
-        _import.groupsUpdatedAt = new Date();
+        _import.newProfileProperties = newProfileProperties;
+        _import.profileUpdatedAt = now;
+        _import.newGroupGuids = newGroupGuids;
+        _import.groupsUpdatedAt = now;
         await _import.save();
       }
 
       let force = false;
 
+      // TODO: did we mean to do something here?
```

Comment:
Oh! please remove this!  We don't need it any more.

https://github.com/grouparoo/grouparoo/pull/938 moved all the of the counting of imports on runs to a task.  It used to increment the count of `profilesImported` here.